aten/src/ATen/native/TensorShape.cpp

```diff
@@ -392,6 +390,90 @@ Tensor cat(TensorList tensors, int64_t dim) {
   return result;
 }
 
+Tensor block_diag(TensorList tensors) {
+  Tensor result;
+  if (tensors.size() == 0) {
+    result = at::empty({0});
```

Comment:
Scipy's returns a tensor of size `(1, 0)` -- does that make sense?
In our documentation we claim that `block_diag` "always returns a 2D tensor", so I think the result here should be 2D at least.
```
[ins] In [4]: import scipy.linalg

[ins] In [5]: scipy.linalg.block_diag()
Out[5]: array([], shape=(1, 0), dtype=float64)
```

Reply:
If this behavior is bizarre we could also just error out when passed an empty list

Reply:
Ah yeah, you're right. Size `(1, 0)` is the right choice.